metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -358,27 +351,75 @@ func CreateStackdriverSink(uri *url.URL) (core.DataSink, error) {
 		}
 	}
 
-	if err := gce_util.EnsureOnGCE(); err != nil {
-		return nil, err
+	var projectId, heapsterZone string
+	// Cluster name and location are required when useNewResourceModel is true.
+	var clusterName, clusterLocation string
+
+	if len(opts["cluster_name"]) >= 1 {
+		clusterName = opts["cluster_name"][0]
 	}
 
-	// Detect project ID
-	projectId, err := gce.ProjectID()
-	if err != nil {
-		return nil, err
+	if len(opts["cluster_location"]) >= 1 {
+		clusterLocation = opts["cluster_location"][0]
 	}
 
-	// Detect zone for old resource model
-	heapsterZone, err := gce.Zone()
-	if err != nil {
-		return nil, err
+	if gce.OnGCE() {
+		// Detect project ID
+		projectId, err = gce.ProjectID()
+		if err != nil {
+			return nil, err
+		}
+
+		// Detect zone for old resource model
+		heapsterZone, err = gce.Zone()
```

Comment:
Is heapsterZone required in case of newResourceModel? Maybe we shouldn't return error in such case?